### PR TITLE
#21375 Stream Graph: Community suggestions

### PR DESCRIPTION
**Pull request by vtkalek**
Axis font-size needs to be changedData

Labels option displays only the measure *field name* and not the measure *value*

Tooltip only shows the group and not measure value

### CODE REVIEW

File: src/settings.ts [hunk 1]
```diff
@@ -38,10 +38,12 @@ module powerbi.extensibility.visual.settings {
 
     export class BaseAxisSettings {
         public static DefaultColor: string = "#777";
+        public static DefaultAxisFontSizeInPoints: string = "8";
```
Comment (uve) [inline src/settings.ts]: use Number type here

File: src/tooltipBuilder.ts [hunk 2]
```diff
@@ -121,6 +121,18 @@ module powerbi.extensibility.visual.tooltipBuilder {
                         value: formattedHighlightedValue
                     });
                 }
+
+                if (seriesData.metadata.values) {
+                    let valuesList: string = "";
+                    for (let sindex: number = 0; sindex < seriesData.metadata.values.length; sindex++) {
```
Comment (uve) [inline src/tooltipBuilder.ts]: can we replace this line by: seriesData.metadata.values.join(" ") ?

File: src/visual.ts [hunk 3]
```diff
@@ -348,6 +345,22 @@ module powerbi.extensibility.visual {
...
+            let textProperties: TextProperties = {
+                text: xMaxValue.toString(),
+                fontFamily: "sans-serif",
+                fontSize: visualSettings.categoryAxis.fontSize + "px"
+            };
+            let xAxisValueMaxSize = textMeasurementService.measureSvgTextWidth(textProperties);
+            let textPropertiesY: TextProperties = {
+                text: yMaxValue.toString(),
+                fontFamily: "sans-serif",
+                fontSize: visualSettings.valueAxis.fontSize + "px"
```
Comment (uve) [inline src/visual.ts]: use PixelConverter.toString(labelFontSize) instead

File: src/visual.ts [hunk 4]
```diff
@@ -442,7 +459,7 @@ module powerbi.extensibility.visual {
             }
 
             this.viewport = StreamGraph.getViewport(options.viewport);
-
+            debugger;
```
Comment (uve) [inline src/visual.ts]: Remove debugger

File: src/visual.ts [hunk 5]
```diff
@@ -499,6 +513,9 @@ module powerbi.extensibility.visual {
 
                 this.behavior.renderSelection(false);
             }
+
+            this.axisX.selectAll("text").style("font-size", this.data.settings.categoryAxis.fontSize);
```
Comment (uve) [inline src/visual.ts]: Move it to renderXAxisLabels() method

File: src/visual.ts [hunk 6]
```diff
@@ -499,6 +513,9 @@ module powerbi.extensibility.visual {
...
+            this.axisY.selectAll("text").style("font-size", this.data.settings.valueAxis.fontSize);
```
Comment (uve) [inline src/visual.ts]: Move it to renderYAxisLabels() method

File: src/visual.ts [hunk 7]
```diff
@@ -989,7 +1018,7 @@ module powerbi.extensibility.visual {
             if (this.data.settings.labels.show) {
                 const labelsXScale: LinearScale<number, number> = d3.scale.linear()
                     .domain([0, series[0].dataPoints.length - 1])
-                    .range([0, width - margin.left - margin.right]);
+                    .range([0, width - margin.left - (this.margin.right + this.data.xAxisValueMaxSize / 2)]);
```
Comment (uve) [inline src/visual.ts]: Please simplify all lines like this one: width - margin.left - (this.margin.right + this.data.xAxisValueMaxSize / 2)

Comment (uve) [reply src/visual.ts]: Also I don't see any unit-tests here

File: src/visual.ts [hunk 8]
```diff
@@ -348,6 +345,22 @@ module powerbi.extensibility.visual {
                 }
             }
 
+            let textProperties: TextProperties = {
+                text: xMaxValue.toString(),
+                fontFamily: "sans-serif",
+                fontSize: visualSettings.categoryAxis.fontSize + "px"
```
Comment (uve) [inline src/visual.ts]: use PixelConverter.toString(labelFontSize) instead

File: src/visual.ts (hunk 5 above)
Comment (vtkalek) [reply src/visual.ts]: It's not working if put to renderXAxisLabels

File: src/visual.ts (hunk 6 above)
Comment (vtkalek) [reply src/visual.ts]: It's not working if put to renderYAxisLabels

File: src/visual.ts [hunk 9]
```diff
@@ -598,16 +618,15 @@ module powerbi.extensibility.visual {
 
                 this.axisY.call(this.yAxisProperties.axis);
                 const yAxisTextNodes: Selection<any> = this.axisY.selectAll("text");
-
-                yAxisTextNodes.style("fill", valueAxisLabelColor);
+                yAxisTextNodes.style("fill", valueAxisLabelColor).style("font-size", this.data.settings.valueAxis.fontSize);
```
Comment (uve) [inline src/visual.ts]: Please move style("font-size") to the next line

File: src/visual.ts [hunk 10]
```diff
@@ -660,7 +676,7 @@ module powerbi.extensibility.visual {
                         transform: StreamGraph.YAxisLabelAngle,
                         fill: valueAxisSettings.labelColor,
                         x: -(marginTop + (height / StreamGraph.AxisLabelMiddle)),
-                        dy: StreamGraph.YAxisLabelDy
+                        y: (StreamGraph.YAxisLabelDy - this.data.yAxisValueMaxTextSize) + "px"
```
Comment (uve) [inline src/visual.ts]: Use PixelConverter for this line

File: src/visual.ts [hunk 11]
```diff
@@ -706,9 +722,8 @@ module powerbi.extensibility.visual {
                 .remove();
 
             const categoryAxisSettings: BaseAxisSettings = this.data.settings.categoryAxis;
-
             this.margin.bottom = categoryAxisSettings.show
-                ? StreamGraph.XAxisOnSize
+                ? StreamGraph.XAxisOnSize + parseInt(this.data.settings.categoryAxis.fontSize.toString())
```
Comment (uve) [inline src/visual.ts]: Try to do not use parseInt(  ...toString()) conversion

File: test/visualTest.ts [hunk 12]
```diff
@@ -341,6 +350,17 @@ module powerbi.extensibility.visual.test {
...
+                        expectedFontSize: string = "22px";
+
+                    (dataView.metadata.objects as any).valueAxis.fontSize = fontSize;
+
+                    visualBuilder.updateFlushAllD3Transitions(dataView);
+
+                    expect($(visualBuilder.yAxisTicks["0"].children["0"].lastChild).css("font-size")).toBe(expectedFontSize);
+                });
             });
         });
```
Comment (uve) [inline test/visualTest.ts]: Add unit-test for Data Label -> Show Values option

**Pull request merged**